src/System.Memory/src/System/SequencePosition.cs

```diff
@@ -13,29 +13,34 @@ namespace System
     /// </summary>
     public readonly struct SequencePosition : IEquatable<SequencePosition>
     {
+        private readonly object _segment;
+        private readonly int _index;
+
         /// <summary>
         /// Creates new <see cref="SequencePosition"/>
         /// </summary>
         public SequencePosition(object segment, int index)
         {
-            Segment = segment;
-            Index = index;
+            _segment = segment;
+            _index = index;
         }
 
         /// <summary>
         /// Segment of memory this <see cref="SequencePosition"/> points to.
         /// </summary>
-        public object Segment { get; }
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public object GetObject() => _segment;
 
         /// <summary>
         /// Index inside segment of memory this <see cref="SequencePosition"/> points to.
         /// </summary>
-        public int Index { get; }
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public int GetInteger() => _index;
```

Comment:
Are we fine with these still being public?

Reply:
Also, can you please add a test similar to https://github.com/dotnet/corefx/issues/27224#issue-297985444 so we are validating correctness here?

For instance, are we making sure we remove the bitmask whenever necessary (ToString, etc).

Reply:
**Question:** Does this comment still hold? https://github.com/dotnet/corefx/pull/27229/files#diff-b484b08a6c445aba9f2da2e98b714a9fR12
```C#
// Properties of this type should not be interpreted by anything but the type that created it.
```

Reply:
> Are we fine with these still being public?

Yes

Reply:
> For instance, are we making sure we remove the bitmask whenever necessary (ToString, etc).

We are not removing anything.

Reply:
Why do these methods need to be public? 

> We are not removing anything.

Should we remove the bitmask in ToString? https://github.com/dotnet/corefx/pull/27229/files#diff-b484b08a6c445aba9f2da2e98b714a9fR62

Otherwise, the index value will not make sense to users.